src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Cancun.cs

```diff
@@ -16,8 +14,66 @@ public partial class EngineRpcModule : IEngineRpcModule
 {
     private readonly IAsyncHandler<byte[], GetPayloadV3Result?> _getPayloadHandlerV3;
 
-    public Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV3(ExecutionPayload executionPayload) =>
-        NewPayload(executionPayload, 3);
+    public Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV3(ExecutionPayload executionPayload, byte[][] blobVersionedHashes)
+    {
+        if (blobVersionedHashes is null)
+        {
+            string error = "Blob versioned hashes must be set";
+            if (_logger.IsWarn) _logger.Warn(error);
+            return ResultWrapper<PayloadStatusV1>.Fail(error, ErrorCodes.InvalidParams);
+        }
+
+        if (executionPayload is null)
+        {
+            string error = "Execution payload must be set";
+            if (_logger.IsWarn) _logger.Warn(error);
+            return ResultWrapper<PayloadStatusV1>.Fail(error, ErrorCodes.InvalidParams);
+        }
+
+        int index = 0;
+        foreach (Transaction tx in executionPayload.GetTransactions())
+        {
+            if (tx.BlobVersionedHashes is null)
+            {
+                continue;
+            }
+
+            foreach (byte[]? blobVersionedHash in tx.BlobVersionedHashes)
+            {
+                if (index == blobVersionedHashes.Length
+                    || blobVersionedHash is null
+                    || blobVersionedHashes[index] is null
+                    || !blobVersionedHash.SequenceEqual(blobVersionedHashes[index]))
+                {
+                    string error = "Blob versioned hashes do not match";
+                    if (_logger.IsWarn) _logger.Warn(error);
+                    return ResultWrapper<PayloadStatusV1>.Success(
+                        new PayloadStatusV1
+                        {
+                            Status = PayloadStatus.Invalid,
+                            LatestValidHash = null,
+                            ValidationError = error
+                        });
+                }
+                index++;
+            }
+        }
+
+        if (index != blobVersionedHashes.Length)
+        {
+            string error = "Blob versioned hashes do not match";
+            if (_logger.IsWarn) _logger.Warn(error);
+            return ResultWrapper<PayloadStatusV1>.Success(
+                new PayloadStatusV1
+                {
+                    Status = PayloadStatus.Invalid,
+                    LatestValidHash = null,
+                    ValidationError = error
+                });
+        }
```

Comment:
Should we move that under `ExcecutionPayload.Validate` extension method?

If we will make versioning change than each ExecutionPayloadVXX should have a validator, that we can pass into `NewPayload` somehow. Those validators can become a chain of validators based on version or a class hierarchy.

If we keep the versioning as is I think this needs to go to centralized point in `ExcecutionPayload.Validate` extension method, where we need to handle all cases and combinations.